pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/worker/WorkerAssignmentStrategy.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.logical.worker;
+
+import org.apache.pinot.query.planner.QueryPlan;
+import org.apache.pinot.spi.annotations.InterfaceStability;
+
+
+/**
+ * WorkerAssignmentStrategy selects the servers to be used for each stage of the query.
+ */
+@InterfaceStability.Evolving
+public interface WorkerAssignmentStrategy {
+  void assignWorkers(QueryPlan queryPlan, long requestId);
+}
```

Comment:
can be address in separate PR but I think we should separate out the WorkerAssignment as a class object --> we shouldn't nested this information inside QueryPlan anymore.

Reply:
Yeah I am not sure what the correct layout is but this definitely feels a bit off. Later we may also choose to instead return a worker assignment instead of modifying the plan in-place